File: ARCADE/Intro/Level 4_5: palindromeRearranging.cs
    using System;
     
    public class Test
    {
    	public static bool palindromeRearranging(string inputString){
            char c; int count; bool cUsed = false;
            char d = '!';
            for (int i = 0; i < inputString.Length; i++){
                c = inputString[i];
                count = 0;
                for (int j = 0; j < inputString.Length; j++){
                    if (inputString[j] == c) count += 1;
                }
                if (count % 2 == 1){
                    if (cUsed == true && d != c){
                        return false;
                    }
                    else{
                        d = c;
                        cUsed = true;
                    }
                }
            }
            return true;
        }
    	public static void Main()
    	{
    		string inputString = "aabb";
    		Console.WriteLine(palindromeRearranging(inputString));
    		string inputString1 = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaabc";
    		Console.WriteLine(palindromeRearranging(inputString1));
    	}
    }

File: ARCADE/Intro/Level 4_5: palindromeRearranging.pas
program palindromeRearranging;

function palindromeRearranging(inputString: string): boolean;
var i, j, count : integer;
    check, rs : boolean;
    c, d : char;
begin
    rs := true;
    check := false; d := '!';
    for i := 1 to length(inputString) do
    begin
        c := inputString[i];
        count := 0;
        for j := 1 to length(inputString) do
            if(c = inputString[j]) then count := count + 1;
        
        if(count mod 2 <> 0) then
            if((check = true) and (d <> c)) then rs := false
            else
            begin
                check := true;
                d := c;
            end;
    end;
    palindromeRearranging := rs;
end;

var inputString : string;
begin
	inputString := 'aabb';
	writeln(palindromeRearranging(inputString));
	inputString := 'abca';
	writeln(palindromeRearranging(inputString));
end.
